src/test/java/StationsTest.java

```diff
@@ -47,14 +47,30 @@ public void checkConnectingStations() throws IOException {
         Stations stations = givenStations();
 
         //when
-        stations.features.get(0).properties.getConnectingStations();
-        List<Integer> connectingStations = stations.features.get(0).properties.connectingStations;
+        stations.features.get(0).properties.findConnectingStations(stations);
+        List<Stations.Station> connectingStations = stations.features.get(0).properties.connectingStations;
 
         //then
         assertEquals("Astor Pl", stations.features.get(0).properties.name);
         assertNotNull(connectingStations);
-        assertTrue(connectingStations.contains(457));
-        assertTrue(connectingStations.contains(105));
+        assertTrue(connectingStations.contains(stations.findStation(457)));
+        assertTrue(connectingStations.contains(stations.findStation(105)));
+    }
+
+    @Test
+    public void checkShortestPath() throws IOException {
+        //given
+        Stations stations = givenStations();
+
+        //when
+        Path path1 = stations.findShortestPath(stations.findStation(32), stations.findStation(105));
```

Comment:
I would make this 3 different tests. And make sure that the stations on the path are correct.